rust/core-lib/src/tabs.rs

```diff
@@ -390,9 +390,23 @@ impl CoreState {
             None => return,
         };
 
-        let ed = &self.editors[&buffer_id];
+        let fin_text = if let Some(mut save_ctx) = self.make_context(view_id) {
+            let editor = save_ctx.editor.borrow();
+            let rope = editor.get_buffer();
+            if save_ctx.pre_save_newline() {
+                let mut text = rope.to_string();
```

Comment:
We should be able to avoid needing to roundtrip to the string, which will be expensive for big documents. Idiomatically this could look like,

```rust
let mut rope = editor.get_buffer().clone();
let len = rope.len();
rope.edit(len.., "\n");
// .. save as normal
```

Reply:
Ah, 'edit' was the function I was looking for :)